src/lib/blueprint.ts

```diff
@@ -397,17 +413,43 @@ const createProperties = (
         type: 'string',
         isDeprecated: false,
         deprecationMessage: '',
+        isUndocumented: false,
       }
     }
 
+    const description =
+      'description' in prop && typeof prop.description === 'string'
+        ? prop.description
+        : ''
+
+    const isUndocumented =
+      'x-undocumented' in prop ? Boolean(prop['x-undocumented']) : false
+    const isDeprecated = 'deprecated' in prop && prop.deprecated === true
+
+    const deprecationMessage =
+      'x-deprecated' in prop && typeof prop['x-deprecated'] === 'string'
+        ? prop['x-deprecated']
+        : ''
+
     const baseProperty = {
       name,
-      description:
-        'description' in prop && typeof prop.description === 'string'
-          ? prop.description
-          : '',
-      isDeprecated: false,
-      deprecationMessage: '',
+      description,
+      isDeprecated,
+      deprecationMessage,
+      isUndocumented,
+    }
+
+    if (baseProperty.description !== '') {
+      const deprecatedMatch = baseProperty.description.match(
+        /---\s*deprecated:(.+?)---/s,
```

Comment:
The yaml should never be in the description, why are we matching here?

Reply:
My bad, I forgot to remove this check